content/security/vulnerabilities.adoc

```diff
@@ -0,0 +1,100 @@
+
+=== Writing Files
+
+When writing, a new plugin could be stored in `$JENKINS_HOME/plugins/`, or a new file in `$JENKINS_HOME/init.groovy.d`, or `$JENKINS_HOME/init.groovy` replaced (link:/doc/book/managing/groovy-hook-scripts/#post-initialization-script-init-hook[documentation]).
+Once Jenkins is restarted, the attacker's code will be executed (CVSS v3.1: `C:H/I:H/A:H`).
+
+=== Reading Files
+
+When reading, the content of files in `$JENKINS_HOME/secrets/` can be used in a variety of ways.
+The following are the most severe known impacts of attackers getting access to the files in this directory:
+
+* **Remote code execution via Resource Root URLs** (Variant 1) +
+    Exploitation requires that all of the following conditions are met:
+    ** The "Resource Root URL" functionality is enabled (see link:/doc/book/security/user-content/#resource-root-url[documentation]).
+    ** The CLI WebSocket endpoint is accessible.
+        This requires that Jenkins is running on a version of Jetty for which Jenkins supports WebSockets.
+        This is the case when using the provided native installers, packages, or the Docker containers, as well as when running Jenkins with the command `java -jar jenkins.war`.
+        Additionally, reverse proxies may not allow WebSocket requests if improperly configured.
+    ** Attackers know, or can guess, the user name of any user with Overall/Read permission.
+* **Remote code execution via Resource Root URLs** (Variant 2) +
+    Exploitation requires that all of the following conditions are met:
+    ** The "Resource Root URL" functionality is enabled (see link:/doc/book/security/user-content/#resource-root-url[documentation]).
+    ** The attacker needs an API token for a (non-anonymous) user account.
+        It is not necessary for this user account to (still) have Overall/Read permission.
+* **Remote code execution via "Remember me" cookie** +
+    Forging a "Remember me" cookie allows attackers to log in to Jenkins using a web browser, thereby gaining access to the Script Console if they forge a cookie for an administrator account.
+    Exploitation requires that all of the following conditions are met:
+    ** The "Remember me" feature is enabled (the default).
+    ** Attackers have Overall/Read permission to be able to read content in files beyond the first few lines.
+* **Remote code execution via stored cross-site scripting (XSS) attacks through build logs** +
+    Forging serialized console note objects allows implementing XSS attacks by injecting arbitrary HTML and JavaScript into build logs.
+    This attack bypasses the protections added for link:/security/advisory/2017-02-01/#persisted-cross-site-scripting-vulnerability-in-console-notes[SECURITY-382 in the 2017-02-01 security advisory].
+    Exploitation requires that attackers can control build log output (e.g., through pull requests).
```

Comment:
```suggestion
    Exploitation requires that attackers can control build log output, such as through pull requests.
```
Suggestion to replace `e.g.` with `such as` per [microsoft style guide](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/e/eg)